visions/accessors.md

```diff
@@ -0,0 +1,348 @@
+    set { ... accept a value ... }
+  }
+}
+```
+
+In this case, the `get` accessor gets compiled into a special kind of method that returns a value, the `set` accessor gets compiled into a method that accepts a new value as an argument.  Reading or writing such a property is compiled into a call to the corresponding method.  Swift provides resilience for stored properties by automatically generating `get` and `set` accessor methods so that clients can be compiled the same way regardless of whether the property is actually written as a stored or computed property.
+
+> The description of “methods” here explains what happens when you access resilient stored properties that are defined in a separate module.  When accessing stored properties from within the *same* module, Swift typically uses efficient direct property access.
+
+However, `get` and `set` accessors have significant limitations.  For example, because the `get` and `set` accessors transfer the value as a method argument or return value, they generally require the value to be copyable, which is no longer universally true in Swift.  This copying also adds significant overhead for large or complex values.  As an additional complication, more complex APIs, such as the subscript access for `Dictionary`, need to construct a temporary value and then clean up that value at the end of the access.
+
+These limitations have led Swift to add a variety of other accessors, including several experimental forms that have never been officially standardized through the Swift Evolution process.  Recent work on noncopyable support and an expanded interest in new kinds of collections has made it important to provide a complete standard set of accessors to support future work.  This document will explain the various accessors that currently exist or might be proposed in the near future and how they will provide the full set of capabilities needed for different applications.
+
+> Note:  Accessors are also used to implement subscripts; the only real difference is that a subscript operation can have additional arguments.  This document will discuss property accesses and subscript operations interchangeably.
+
+## Two Dimensions of Access
+
+There are a wide variety of theoretically possible accessors, varying in what kind of access they provide and what mechanism they use to provide it.  Different programming languages support different combinations.  For Swift, there are three fundamental kinds of access that we are interested in:
+
+* Read access.
+* Write or “set” access.
+* Update access.
+
+Write and update access differ in that write access can create a value that did not exist before, while update access requires there be a pre-existing value.  For example, consider how the dictionary write operation `myDictionary["newKey"] = value` can create a value that did not exist before. In contrast, `myDictionary["existingKey"] += 1` modifies a value that must necessarily already exist.
```

Comment:
Swift's actual `Dictionary` indeed does not quite support this, but a simple property misses the key point:  A pure write operation can create/initialize a stored value where no stored value existed before.  A simple property always has storage associated and by Swift's definite-initialization rules that storage must always be initialized.

I've changed the text here to refer more vaguely to "write operation on an expandable collection" to try to avoid this problem.